features/timeseries/src/main/java/org/opennms/netmgt/timeseries/meta/TimeSeriesMetaDataDao.java

```diff
@@ -0,0 +1,168 @@
+ * This file is part of OpenNMS(R).
+ *
+ * Copyright (C) 2006-2016 The OpenNMS Group, Inc.
+ * OpenNMS(R) is Copyright (C) 1999-2016 The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is a registered trademark of The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ *
+ * OpenNMS(R) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with OpenNMS(R).  If not, see:
+ *      http://www.gnu.org/licenses/
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.netmgt.timeseries.meta;
+
+import static org.opennms.netmgt.timeseries.integration.support.TimeseriesUtils.toResourceId;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+
+import javax.sql.DataSource;
+
+import org.opennms.core.utils.DBUtils;
+import org.opennms.integration.api.v1.timeseries.StorageException;
+import org.opennms.netmgt.model.ResourcePath;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
+
+/**
+ * TimeSeriesMetaDataDao stores string values associated with resourceids in the database. It leverages a Guava cache.
+ * Design choice: for speed and high throughput we don't keep the cache and the database 100% in sync at the time of writing.
+ * Latest 5 minutes after writing they will be the same.
+ */
+@Service
+public class TimeSeriesMetaDataDao {
+
+    private static final Logger LOG = LoggerFactory.getLogger(TimeSeriesMetaDataDao.class);
+
+    private final DataSource dataSource;
+
+    private final Map<String, Map<String, String>> cache; // resourceId, Map<name, value>
+
+    @Autowired
+    public TimeSeriesMetaDataDao(final DataSource dataSource) {
+        this.dataSource = dataSource;
+        Cache<String, Map<String, String>> cache = CacheBuilder.newBuilder()
+                .maximumSize(10000)
```

Comment:
Should be able to tune the cache settings.

Reply:
done.